### PR TITLE
vimPlugins.coc-prettier: Add override

### PR DESCRIPTION
**Pull request by ersinakinci**
###### Motivation for this change

This PR adds an override for `vimPlugins.coc-prettier` as a partial fix for https://github.com/NixOS/nixpkgs/issues/64560. ~Depends on https://github.com/NixOS/nixpkgs/pull/72501.~ Previously depended on https://github.com/NixOS/nixpkgs/pull/72501, which has now been squashed into this PR.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @rvolosatovs

### CODE REVIEW

**Comment by jonringer** [comment]
since the 2 commits are necessary to perform the fix, do you mind squashing them both with the commit message: 
```
vimPlugins.coc-prettier: Add node dependency
```

**Comment by earksiinni** [comment]
@jonringer done!

**Review by rvolosatovs — APPROVED** [review]

**Comment by jonringer** [comment]
@GrahamcOfBorg build vimPlugins.coc-prettier

**Comment by jonringer** [comment]
hmm, im just not sure about the several thousand node packages bump ... :(

**Comment by rvolosatovs** [comment]
@jonringer I believe that's a side effect of node package packaging approach: https://github.com/NixOS/nixpkgs/blob/95dfbe2d63/doc/languages-frameworks/node.section.md#L37-L47

**Comment by earksiinni** [comment]
@jonringer, what @rvolosatovs said. I also ran this by @adisbladis in the #nixos IRC channel and they confirmed that this is normal and even desirable.

I think there's a broader conversation to be had about plugin/package management policy as the "updates a million things" behavior is the standard workflow for plugins/packages in other Nixpkgs subsystems (e.g., Vim, Ruby). But in the context of this particular PR, this is actually the blessed path.

**Comment by earksiinni** [comment]
Any progress on this?

File: pkgs/development/node-packages/node-packages-v12.nix [hunk 1]
```diff
@@ -2047,13 +2047,13 @@ let
         sha512 = "w2VwSrBoHa5BsSyH+KxEqeQBAllHhccyMFVHtGtdMpF4W7IRWfZjFiQceJPChOeTsSDVUpER2T8FA93pr0L+QA==";
       };
     };
-    "temp-0.9.0" = {
+    "temp-0.9.1" = {
```
Comment (jonringer) [inline pkgs/development/node-packages/node-packages-v12.nix]: I think this change is unrelated, you added a package to the v10 list, but an entry was updated in the v12 list

File: pkgs/misc/vim-plugins/overrides.nix [hunk 2]
```diff
@@ -10,6 +10,7 @@
 , languagetool
 , Cocoa, CoreFoundation, CoreServices
 , buildVimPluginFrom2Nix
+, nodePackages
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/overrides.nix]: This change should be squashed with the entry that uses it change

File: pkgs/development/node-packages/node-packages-v10.json [hunk 3]
```diff
@@ -16,6 +16,7 @@
 , "browserify"
 , "castnow"
 , "clean-css"
+, "coc-prettier"
```
Comment (jonringer) [inline pkgs/development/node-packages/node-packages-v10.json]: shouldn't that cause a change in pkgs/development/node-packages/node-packages-v10.nix, instead it generates a new file, which seems weird to me.

File: pkgs/misc/vim-plugins/overrides.nix [hunk 4]
```diff
@@ -134,6 +135,12 @@ self: super: {
     };
   };
 
+  coc-prettier = buildVimPluginFrom2Nix {
+    pname = "coc-prettier";
+    version = nodePackages.coc-prettier.version;
+    src = nodePackages.coc-prettier + "/lib/node_modules/coc-prettier";
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/overrides.nix]: not sure if this is right, but I think you were trying to create a path.
```suggestion
    src = "${nodePackages.coc-prettier}/lib/node_modules/coc-prettier";
```

**Review by jonringer — COMMENTED** [review]
Please also change the commit history so that it's not broken if at a particular commit (you reference nodePackages.coc-prettier before it was added)

the history should reflect:
```
nodePackages.coc-prettier: init
vimPlugins.overrides: add coc-prettier
neovim: use TMPDIR as home during initilizaiton
```

**Comment by FelschR** [comment]
Any update on this?

**Comment by earksiinni** [comment]
Hey all, I got distracted from finishing this up. I'm going to incorporate @jonringer's suggestions and clean up this PR, hopefully we can get this merged soon.

**Comment by earksiinni** [comment]
@jonringer, I've cleaned up this PR and incorporated your suggestions from November. Let me know what you think.

**Comment by earksiinni** [comment]
Friendly ping 🙂.

If we can merge this, I have a bunch of other coc extensions that I want to merge, as well. I've successfully been using this approach for ~10 extensions on my local machine via overlays.

File: pkgs/development/node-packages/node-packages-v10.json (hunk 3 above)
Comment (earksiinni) [reply pkgs/development/node-packages/node-packages-v10.json]: Just as a reminder from November, I agree that this is weird, but apparently regenerating the entire node-packages-v10.nix and updating a ton of NPM packages in the process is the blessed path. I'm basing this off of 1) a conversation I had in the #nixos IRC channel and 2) the documentation.

**Pull request merged**